db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/ExecutePicked.java

```diff
@@ -78,22 +77,20 @@ private void executePickedExecution(Execution execution) {
         }
 
         Instant executionStarted = clock.now();
-        try {
-            LOG.debug("Executing " + execution);
-            CompletionHandler completion = task.get().execute(execution.taskInstance, new ExecutionContext(schedulerState, execution, schedulerClient));
+        CompletableFuture<CompletionHandler> completableFuture = task.get().execute(execution.taskInstance, new ExecutionContext(schedulerState, execution, schedulerClient));
+        completableFuture.whenComplete((completion, ex) -> {
```

Comment:
does this whencomplete execute on the executor ?

Reply:
According to my test, the whole chain inside completableFuture will run on the same thread. So if the future returned by execute() does not run on the executor, whencomplete will not neither.